packages/frontend/src/utils/getVolunteers.ts

```diff
@@ -1,68 +1,114 @@
 import { firebaseDb } from 'src/main';
 import { IVolunteer } from 'uva-shared/src/types/volunteer';
-import {
-  collection,
-  getDocs,
-  limit,
-  orderBy,
-  query,
-  startAfter,
-  where,
-  DocumentSnapshot,
-  or,
-  and,
-} from 'firebase/firestore';
+import { collection, getDocs, limit, orderBy, query, startAfter, where, DocumentSnapshot } from 'firebase/firestore';
 import { QueryArgs } from './getEvents';
 
 export interface GetVolunteersResult {
   readonly volunteers: IVolunteer[];
   readonly lastVolunteer: DocumentSnapshot | null;
 }
 
+const formatDateToDDMMYYYY = (dateString: string): string => {
+  if (!dateString) return "";
+  const [year, month, day] = dateString.split('-');
+  return `${day}-${month}-${year}`;
+};
+
 export const getVolunteers = async (
   name?: string,
+  languages: string[] = [],
+  volunteerAgeFrom = '',
+  volunteerAgeTo = '',
+  gender = '',
+  // After adding ability of blocking volunteer, open this block
+  // showBlocked = false,
   lastVolunteer?: DocumentSnapshot | null,
   pageSize = 20,
 ): Promise<GetVolunteersResult> => {
-  let volunteers: IVolunteer[] = [];
-  let querySnapshot;
-
-  const searchByField = async () => {
-    const queryArgs: QueryArgs[] = [limit(pageSize)];
-
-    let queryForVolunteers;
-    if (name) {
-      const start = name.toLowerCase();
-      const end = start + '\uf8ff';
-      queryForVolunteers = query(
-        collection(firebaseDb, FirebaseCollection.Volunteers),
-        or(
-          and(where('firstName', '>=', start), where('firstName', '<', end)),
-          and(where('lastName', '>=', start), where('lastName', '<', end)),
-        ),
-        orderBy('firstName', 'desc'),
-      );
-    } else {
-      queryForVolunteers = query(collection(firebaseDb, FirebaseCollection.Volunteers), ...queryArgs);
-    }
-
-    if (lastVolunteer) {
-      queryArgs.push(startAfter(lastVolunteer));
-    }
-    return await getDocs(queryForVolunteers);
-  };
+  const queryArgs: QueryArgs[] = [];
+  const formattedAgeFrom = formatDateToDDMMYYYY(volunteerAgeFrom);
+  const formattedAgeTo = formatDateToDDMMYYYY(volunteerAgeTo);
+
+  if (gender) {
+    queryArgs.push(where('gender', '==', gender));
+  }
+
+  // After adding ability of blocking volunteer, open this block
+  // if (showBlocked) {
+  //   queryArgs.push(where('isBlocked', '==', true));
+  // }
+
+  if (volunteerAgeFrom && volunteerAgeTo) {
+    queryArgs.push(where('birthDate', '<=', formattedAgeTo), where('birthDate', '>=', formattedAgeFrom));
+  } else if (volunteerAgeFrom) {
+    queryArgs.push(where('birthDate', '>=', formattedAgeFrom));
+  } else if (volunteerAgeTo) {
+    queryArgs.push(where('birthDate', '<=', formattedAgeTo));
+  }
+
+  if (languages.length > 0) {
+    queryArgs.push(where('language', 'array-contains-any', languages));
+  }
+
+  if (lastVolunteer) {
+    queryArgs.push(startAfter(lastVolunteer));
+  }
+
+  let queryForVolunteers;
+  if (name) {
```

Comment:
there sholdn't be query and return, just
```
if (name) {
      const start = name.toLowerCase();
      const end = start + '\uf8ff';
      queryArgs.push(or(
          and(where('firstName', '>=', start), where('firstName', '<', end)),
          and(where('lastName', '>=', start), where('lastName', '<', end)),
        ),
    );
}
queryForVolunteers = query(
      collection(firebaseDb, FirebaseCollection.Volunteers),
      ...queryArgs,
      orderBy('firstName', 'asc'),
      limit(pageSize),
    );
```

Reply:
it's needed to be added to query, user can have both filters and search string at the same time